File: hastane_menu/lib/models/hospital_info.dart
import 'package:hastane_menu/core/constants/app_config.dart';

/// Yemekhane / hastane bilgisi (Bilgi sayfası).
///
/// ⚠️ Turkcell HBYS dokümanında hastane bilgisi ucu YOKTUR ve yönetici paneli
/// kaldırılmıştır → bu bilgiler tamamen [AppConfig]'ten gelir (tek merkez).
class HospitalInfo {
  const HospitalInfo({
    required this.hospitalName,
    required this.subtitle,
    required this.description,
    required this.workingHours,
    required this.location,
    required this.contact,
  });

  final String hospitalName;
  final String subtitle;
  final String description;
  final String workingHours;
  final String location;
  final String contact;

  /// Config'ten üretilen tek kaynak.
  factory HospitalInfo.fromConfig() => const HospitalInfo(
    hospitalName: AppConfig.hospitalName,
    subtitle: AppConfig.appSubtitle,
    description: AppConfig.cafeteriaDescription,
    workingHours: AppConfig.workingHours,
    location: AppConfig.location,
    contact: AppConfig.contact,
  );
}

File: hastane-api/Domain/Entities/HospitalInfo.cs
using Ozi.Domain.Shared;

namespace Ozi.Domain.Entities;

/// <summary>
/// Hastane / yemekhane bilgileri (tek satır). Panel'den düzenlenir, mobil "Bilgi"
/// sayfası <c>GET /hospital-info</c> ile okur. Alanlar mobil <c>AppConfig</c> ile eşleşir.
/// </summary>
public class HospitalInfo : EntityBase
{
    public string HospitalName { get; set; } = string.Empty;
    public string Subtitle { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string WorkingHours { get; set; } = string.Empty;
    public string Location { get; set; } = string.Empty;
    public string Contact { get; set; } = string.Empty;
}
